constraints.go

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+package magistrala
+
+type Constraints struct {
+	Users        uint32 `toml:"users"`
+	Domains      uint32 `toml:"domains"`
+	Things       uint32 `toml:"things"`
+	Groups       uint32 `toml:"groups"`
+	Channels     uint32 `toml:"channels"`
+	MsgRateLimit uint32 `toml:"msg_rate_limit"`
+}
+
+// ConstraintsProvider specifies an API for obtaining entity constraints.
+type ConstraintsProvider interface {
+	// Constraints returns constraints for the entities.
+	Constraints() (Constraints, error)
+}
```

Comment:
```suggestion
// Copyright (c) Abstract Machines
// SPDX-License-Identifier: Apache-2.0

package magistrala

type Operation int

const (
	Create Operation = iota
)

// Constraints specifies an API for obtaining entity constraints.
type Constraints interface {
	// Constraints returns constraints for the entities.
	CheckLimits(operation Operation) error
}
```

something like this expected